spec/system/users_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  scenario 'displays the username, profile picture, and number of posts for each user' do
+    user = User.create(name: 'Lilly', posts_counter: 3, photo: 'https://randomuser.me/api/portraits/women/70.jpg',
+                       bio: 'Teacher from Poland.')
+
+    visit user_path(user)
+
+    profile_picture = find('img')
+
+    expect(page).to have_content(user.name)
+    expect(profile_picture).to be_visible
+    expect(profile_picture['src']).to eq user.photo
+    expect(page).to have_content(3)
+  end
+
+  scenario 'redirects to user show page when clicking on a user' do
+    user = User.create(name: 'Lilly', posts_counter: 4, photo: 'https://randomuser.me/api/portraits/women/70.jpg',
+                       bio: 'Teacher from Poland.')
+    user_show_page_url = "#{Capybara.app_host}/users/#{user.id}"
+
+    visit users_path(user)
+
+    within('.user-details') do
+      click_link 'Lilly'
+    end
+
+    expect(page).to have_current_path(user_show_page_url)
+  end
```

Comment:
The tests for the user index page should be in its own file and that file should test only the user index page to follow the standard best practices.

Reply:
@jerryowusu can you kindly check this, it is optional to put index on its file, here am testing the view, check this first instruction 'Use Capybara to write integration tests for each view in your project.'

Reply:
Sorry Tracy-miranja, this cannot be made optional since it is part of the project requirements and it helps to achieve best standard practices.